cap/modules/records/minters.py

```diff
@@ -33,7 +33,12 @@
 def cap_record_minter(record_uuid, data):
     """Mint record identifiers."""
     assert 'control_number' not in data
-    pid_value = generate_recid(data['_experiment'])
+
+    # FASER analysis have same PID before and after publishing.
+    if data['_experiment'] == 'FASER':
```

Comment:
this logic should happen inside `generate_recid` that we are passing the experiment